[script.transmission] 1.0.0

basilgello opened the pull request:
### Description
The control panel add-on for Transmission Bittorrent client for Kodi 19.0 "Matrix" and newer versions

This repository is a rework of original Transmission-XBMC addon by Correl J. Roush for Kodi 19.0 "Matrix".

Since version 19.0 "Matrix", the Python 2 support will be dropped and only Python 3 addons will be supported.

Also, the original addon featured the 'Search Torrent' functionality using pirate sites. This violates the rules of official Kodl respository, so this addon has only 'Add Torrent File' functionality.
### Checklist:
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

enen92 commented on script.transmission/README.md:
```diff
@@ -0,0 +1,26 @@
+Transmission-XBMC
```
please change to Transmission for Kodi (this naming is not allowed as it violates our trademark)

enen92 commented on script.transmission/README.md:
```diff
@@ -0,0 +1,26 @@
+Transmission-XBMC
+=================
+
+A client for the popular [Transmission](http://www.transmissionbt.com/) BitTorrent application for [XBMC](http://xbmc.org/)
```
kodi?

basilgello replied:
Done

enen92 commented on script.transmission/README.md:
```diff
@@ -0,0 +1,26 @@
+Features
+--------
+
+Currently, Transmission-XBMC supports viewing, adding, removing, starting and stopping torrents. More advanced features may be added in future releases.
```
same here

basilgello replied:
done

enen92 commented on script.transmission/README.md:
```diff
@@ -0,0 +1,26 @@
+
+This script is intended for Kodi 19 (Matrix).
+
+This script can be installed via the addon manager within XBMC. Attempting to install it manually may not work, as it requires the simplejson library which is automatically installed by the addon manager.
```
kodi

basilgello replied:
done

enen92 commented on script.transmission/README.md:
```diff
@@ -0,0 +1,26 @@
+License
+-------
+
+Transmission-XBMC is licensed under the terms of the [MIT license](http://www.opensource.org/licenses/mit-license.html).
```
same

basilgello replied:
done

enen92 commented on script.transmission/addon.xml:
```diff
@@ -0,0 +1,43 @@
+    <import addon="script.module.beautifulsoup4" version="4.8.2" />
+    <import addon="script.module.kodi-six" version="0.1.3"/>
+    <import addon="script.module.requests" version="2.22.0"/>
+    <import addon="script.module.simplejson" version="3.17.0"/>
```
not needed anymore

basilgello replied:
Removing these imports complaints on missing kodi_six. Or do you mean kodi-six should not be used here?

enen92 replied:
I mean any reference to simplejson, this was only for helix and below

enen92 commented on script.transmission/default.py:
```diff
@@ -0,0 +1,22 @@
+__settings__ = xbmcaddon.Addon(id='script.transmission')
+__language__ = __settings__.getLocalizedString
+
+BASE_RESOURCE_PATH = xbmc.translatePath( os.path.join( __settings__.getAddonInfo('path'), 'resources', 'lib' ) )
```
please use relative imports and not `sys.path.append`

enen92 commented on script.transmission/default.py:
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
```
this file should be as small as possible, only used to import your main codebase, propagate `sys.argv` and run. Reason is this is the only file that is not compiled to bytecode by kodi's python interpreter.

You should consider do something like:

```python
from sys import argv
from resources.lib import myscript

myscript.run(argv)
```

enen92 commented on script.transmission/resources/lib/player.py:
```diff
@@ -0,0 +1,60 @@
+__settings__ = xbmcaddon.Addon(id='script.transmission')
+
+BASE_RESOURCE_PATH = xbmc.translatePath( os.path.join( __settings__.getAddonInfo('path'), 'resources', 'lib' ) )
+sys.path.append (BASE_RESOURCE_PATH)
```
same

enen92 commented on script.transmission/resources/lib/player.py:
```diff
@@ -0,0 +1,60 @@
+        self.prev_settings = {}
+        self.refreshSettings()
+
+    def onPlayBackStarted(self):
```
replaced by `onAvStarted` in leia

enen92 commented on script.transmission/resources/lib/player.py:
```diff
@@ -0,0 +1,60 @@
+
+    def onPlayBackStarted(self):
+        self.refreshSettings()
+        if self.active and xbmc.Player().isPlayingVideo():
```
It is not guaranteed that when this event is called (player initialization) player is already playingvideo, another reason to change it to onAVStarted

enen92 commented on script.transmission/resources/lib/search.py:
```diff
@@ -0,0 +1,129 @@
+    def search(terms):
+        return NotImplemented
+
+class TPB(Search):
```
Oh well, not allowed at all!

enen92 replied:
Drop any piracy related functionality from the addon if you want it on our repo.

enen92 commented on script.transmission/resources/lib/search.py:
```diff
@@ -0,0 +1,129 @@
+                'leechers': leechers,
+            })
+        return torrents
+class Kickass(Search):
```
...

enen92 commented on script.transmission/resources/lib/search.py:
```diff
@@ -0,0 +1,129 @@
+            else:
+                raise
+        return torrents
+class L337x(Search):
```
...

enen92 commented on script.transmission/resources/lib/search.py:
```diff
@@ -0,0 +1,129 @@
+                'leechers': leechers,
+            })
+        return torrents
+class Lime(Search):
```
...

enen92 commented on script.transmission/resources/skins/Default/skin.xml:
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
remove this file from the submission, this is no longer used (and never was)

enen92 commented on script.transmission/README.md:
```diff
@@ -0,0 +1,26 @@
+
+This script is intended for Kodi 19 (Matrix).
+
+This script can be installed via the addon manager within XBMC. Attempting to install it manually may not work, as it requires the simplejson library which is automatically installed by the addon manager.
```
remove simplejson

enen92 commented:
@basilgello for your information: https://github.com/xbmc/repo-scripts/pull/1455

Sorry, we have to do it. You're free to submit an update that drops the search functionality on piracy sites.

basilgello commented:
> @basilgello for your information: #1455
> 
> Sorry, we have to do it. You're free to submit an update that drops the search functionality on piracy sites.

OK. I will drop the search functionality now.

enen92 commented on script.transmission/README.md:
```diff
@@ -0,0 +1,26 @@
+Transmission-Kodi
```
transmission-kodi violates our trademark policy...

Please rename it to just Transmission, Transmission Client, or "Transmission for Kodi or whatever that doesn't cause confusion as the source.

basilgello replied:
Transmission Control Panel sounds right?

enen92 replied:
Sure :)

enen92 commented on script.transmission/README.md:
```diff
@@ -0,0 +1,26 @@
+License
+-------
+
+Transmission-Kodi is licensed under the terms of the [MIT license](http://www.opensource.org/licenses/mit-license.html).
```
same here

enen92 commented on script.transmission/README.md:
```diff
@@ -0,0 +1,26 @@
+Features
+--------
+
+Currently, Transmission-Kodi supports viewing, adding, removing, starting and stopping torrents. More advanced features may be added in future releases.
```
same here

enen92 commented on script.transmission/addon.xml:
```diff
@@ -0,0 +1,42 @@
+    <description lang="uk_UA">Transmission-Kodi дозволяє переглядати список торентів, додавати, видаляти, стартувати та зупиняти торенти.</description>
+    <platform>all</platform>
+    <license>MIT</license>
+    <forum></forum>
```
remove forum element if empty (this is parsed by our wiki and addon websites)

enen92 commented on script.transmission/addon.xml:
```diff
@@ -0,0 +1,42 @@
+    <summary lang="ca_ES">Un client de Transmission, la popular aplicació de Bittorrent</summary>
+    <summary lang="ru_RU">Панель управления для популярного торрент-клиента Transmission</summary>
+    <summary lang="uk_UA">Панель керування для популярного торент-кліента Transmission</summary>
+    <description lang="en_GB">Transmission-Kodi supports viewing, adding, removing, starting and stopping torrents.</description>
```
remove Transmission-Kodi and use just transmission (or similar)

basilgello commented:
@enen92 I have refactored the code and tested it. Now adding torrent files works but piracy search is removed. Also I cleaned some dependencies and removed spaghetti code like sys.path manipulation.

Please have a look again and thanks for previous 20+ comments :)
